exporter.py

```diff
@@ -85,70 +194,8 @@ def start_exporter_server(port):
         start_http_server(port)
         print("Exporter is running...")
     except Exception as e:
-        logger.critical('Error while opening port: {}', format(e))
-        print(e)
-
-
-def check_nitro_access(protocol, nsip, username, password, ns_cert):
-    '''Validates if exporter is able access ADC.'''
-
-    url = '%s://%s/nitro/v1/config/' % (protocol, nsip)
-    try:
-        response = requests.get(url, verify=ns_cert, auth=HTTPBasicAuth(username, password))
-        if (response.status_code == requests.status_codes.codes.unauthorized):
-            logger.error('Invalid username or password for Citrix Adc!, Unaurthorized Err : {}'.format(response.status_code))
-            return False
-    except requests.exceptions.RequestException as err:
-        logger.error('Nitroc Access Error {}'.format(err))
-        return False
-    except Exception as e: 
-        logger.error("Unable to authenticated ADC nitro credentials {}".format(e))
-        return False
-    return True
-
-
-def get_sslcertkey_stats(protocol, nsip, username, password, nitro_timeout, ns_cert):
-    '''Validates if exporter is able fetch stats access from ADC when it's fully configured.'''
-
-    headers = {'X-NITRO-USER': username, 'X-NITRO-PASS': password}
-    url = '%s://%s/nitro/v1/config/sslcertkey' % (protocol, nsip)
-    try:
-        r = requests.get(url, headers=headers, verify=ns_cert, timeout=nitro_timeout)
-        data = r.json()
-        if data['sslcertkey'] is None:
-            return False
-        else:
-            return True
-    except Exception as e:
-        logger.warning("Unable to access stats, ADC still not fully configured")
-        return False
-    return True
-
-
-def verify_ns_session_access(nsip, ns_protocol, ns_user, ns_password, ns_cert):
```

Comment:
There's a reason for these methods to be here. When exporter is side car to CPX, then CPX take a while to come up, most of the retry time that you set will get elapsed while CPX is not even UP. There's no way to know if we are monitoring CPX or VPX, so these initial checks were put. If you have an alternative to that, then, sugget.

Reply:
what i would do, is keep these methods with a separate retry decorator, for a given timeframe, if it succeeds them its fine, else, exit.

Reply:
for this issue i think we should just continue to try in a well behaved (tm) way (e.g. backoff, timeouts) and return the up metric as 0. I'm thinking about it on how to implement it in a good way.